surf/src/diff.rs

```diff
@@ -88,6 +88,19 @@ pub struct CopyFile {
     pub new_path: Path,
 }
 
+#[cfg_attr(
+    feature = "serialize",
+    derive(Serialize),
+    serde(rename_all = "camelCase")
+)]
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum EofNewLine {
+    NoneMissing,
```

Comment:
```suggestion
```

Since this is used with an `Option` we don't need this case. I got tipped off when I looked below and you were always calling the function with `Some` :)

Reply:
Indeed, and then we could change this type to `MissingNewline { Old, New, Both }` perhaps

Reply:
It's not always called with `Some`, example [here](https://github.com/radicle-dev/radicle-surf/pull/201/files#diff-6ff820339b0f953493d4229a5daceea07f09a7b92ed4677a89266f9174941ab8R291) and [here](https://github.com/radicle-dev/radicle-surf/pull/201/files#diff-6ff820339b0f953493d4229a5daceea07f09a7b92ed4677a89266f9174941ab8R448). Having `NoneMissing` would mean it was processed and being `None` means it was not processed, as we also need to process cases of file addition and deletion (which is not coverered here, in `ModifiedFile`).

I can change it if you still want?

Reply:
As I pointed out in my other comment one of those cases is essentially dead code so let's discount it. For the binary case, my feeling is that the consumer is just going to ignore `eof` anyway, so I think my point still stands. However, I'm not sure I understand your point about addition and deletion -- I thought this is only regarding modified files?

Reply:
Fixed here https://github.com/radicle-dev/radicle-surf/commit/53eb72f32cf9478248d5322b8210fe20c03a4368#diff-6ff820339b0f953493d4229a5daceea07f09a7b92ed4677a89266f9174941ab8R97-R101